Update version of commons-lang to 3.5

Mats-SX opened the pull request.

Mats-SX committed:
Update version of commons-lang to 3.5

MishaDemianenko commented on pom.xml:
```diff
@@ -584,7 +584,7 @@
         <groupId>org.apache.commons</groupId>
         <artifactId>commons-lang3</artifactId>
-        <version>3.3.2</version>
+        <version>3.5</version>
```
since we do an update, shouldn't we consider latest stable 3.6? or you need a specific version for reasons?

Mats-SX replied:
I'm interested in `3.5` because Spark uses that: https://github.com/apache/spark/blob/master/pom.xml#L173 and we're using the neo4j frontend from our Spark-dependent project.

MishaDemianenko replied:
ok, thx!

Mats-SX committed:
Use WordUtils from commons-text

Arguments depended on a buggy implementation of WordUtils.wrap() where
leading whitespace on new lines was not stripped despite the docs saying it should.
In commons-lang3 3.6 WordUtils is deprecated and moved to another module
commons-text. This commit introduces that module and solves the leading-space
problem for lines that do not need to be wrapped by not calling WordUtils.wrap
for them. Lines that need wrapping and start with leading whitespace
(indentation) will still have it stripped going forward.

Mats-SX commented:
Build failure is related to downstream `NOTICE.txt` (docs module) not containing the new dependency. Should not hinder this PR from getting through -- discussing options with docs team.

EDIT: Closed by mistake: reopening.
